main.py

```diff
@@ -0,0 +1,69 @@
+from fastapi import FastAPI, HTTPException
+from fastapi.params import Depends
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from database import SessionLocal
+
+app = FastAPI()
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get("/authors/", response_model=list[schemas.Author])
+def read_all_authors(db: Session = Depends(get_db)):
+    return crud.get_all_authors(db)
+
+
+@app.post("/authors/", response_model=schemas.Author)
+def create_author(
+    author: schemas.AuthorCreate,
+    db: Session = Depends(get_db)
+):
+    return crud.create_author(db=db, author=author)
+
+
+@app.get("/authors/{author_id}/", response_model=schemas.Author)
+def read_single_author(
+    author_id: int,
+    db: Session = Depends(get_db)
+):
+    db_author = crud.get_author_by_id(db=db, author_id=author_id)
+
+    if not db_author:
+        raise HTTPException(
+            status_code=404,
+            detail="Author not found"
+        )
+
+    return db_author
+
+
+@app.get("/books/", response_model=list[schemas.Book])
+def read_all_books(
+    author_id: int | None = None,
```

Comment:
The `author_id` parameter is typed as `int | None`, which is correct and consistent with the database model. Ensure that this matches the type used in the `crud.get_all_books` function to avoid type inconsistencies.